lisa/analysis/idle.py

```diff
@@ -236,6 +236,32 @@ def df_cluster_idle_state_residency(self, cluster):
         idle_time_df.index.name = 'idle_state'
         return idle_time_df
 
+    @requires_events('cpu_idle')
+    def idle_cpus_at(self, timestamp):
+        """
+        :returns: list of CPUs that were idle at the given ``timestamp``
+        """
+        idles = []
+        for cpu in range(self.trace.cpus_count):
+            df_active = trace.analysis.idle.signal_cpu_active(cpu)
```

Comment:
AFAICT this is the equivalent of only using `df_prev_index()`.

It might not be super likely, but you could have started the trace with a certain CPU busy, and it only goes in idle much later in the trace. That means you have a chunk of your trace where there is no previous index for any requested timestamp, but you can still figure out if the CPU was idle by looking at the next index (which is what I do).

Reply:
yes, that issue should be solved at the "signal" level rather than bloating the rest of LISA with partial workarounds. Even when considering the next index, the issue will still creep up if you use a small window on the trace. Once the signal fixup is in place, all these issues will disappear, so I'd rather keep the code clean so we don't end up reimplementing/simplifying dozens of helpers just because we fixed the root issue.